util/azure/utils.go

```diff
@@ -0,0 +1,341 @@
+	env := azure.PublicCloud
+	var err error
+	if environment != "" {
+		env, err = azure.EnvironmentFromName(environment)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse environment for Azure.")
+		}
+	}
+
+	endpoint := ""
+
+	switch clusterType {
+	case ConnectedClusters:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatARC, env.ResourceManagerEndpoint)
+	case ManagedClusters:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	case Fleets:
+		endpoint = fmt.Sprintf(OperationsEndpointFormatAKS, env.ResourceManagerEndpoint)
+	default:
+		return nil, errors.Errorf("Failed to create endpoint for Get Operations call. Cluster type %s is not supported.", clusterType)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create request for Get Operations call.")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("guard-%s-%s-%s", v.Version.Platform, v.Version.GoVersion, v.Version.Version))
+
+	var token string
+	if clusterType == ConnectedClusters {
+		tokenProvider := graph.NewClientCredentialTokenProvider(clientID, clientSecret,
+			fmt.Sprintf("%s%s/oauth2/v2.0/token", env.ActiveDirectoryEndpoint, tenantID),
+			fmt.Sprintf("%s/.default", env.ResourceManagerEndpoint))
+
+		authResp, erro := tokenProvider.Acquire("")
+		if erro != nil {
+			return nil, errors.Wrap(erro, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = authResp.Token
+	} else { // AKS and Fleet
+		tokenProvider := graph.NewAKSTokenProvider(loginURL, tenantID)
+
+		authResp, err := tokenProvider.Acquire("")
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting authorization headers for Get Operations call.")
+		}
+
+		token = authResp.Token
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	client := httpclient.DefaultHTTPClient
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to send request for Get Operations call.")
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error in reading response body")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("Request failed with status code: %d and response: %s", resp.StatusCode, string(data))
+	}
+
+	operationsList := OperationList{}
+	err = json.Unmarshal(data, &operationsList)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to decode response")
+	}
+
+	var finalOperations []Operation
+	for _, op := range operationsList.Value {
+		if *op.IsDataAction && strings.Contains(op.Name, clusterType) {
+			finalOperations = append(finalOperations, op)
+		}
+	}
+
+	printFinalOperations, _ := json.Marshal(finalOperations)
```

Comment:
let's only do the marshaling when v >= 5

Reply:
done